example/generator/src/SinewaveTelemetryProvider.js

```diff
@@ -24,96 +24,47 @@
 /**
  * Module defining SinewaveTelemetryProvider. Created by vwoeltje on 11/12/14.
  */
-define(
-    ["./SinewaveTelemetrySeries"],
-    function (SinewaveTelemetrySeries) {
-        "use strict";
+define([
+    "./SinewaveTelemetrySeries",
+    "./GeneratorProvider"
+], function (
+    SinewaveTelemetrySeries,
+    GeneratorProvider
+) {
 
-        /**
-         *
-         * @constructor
-         */
-        function SinewaveTelemetryProvider($q, $timeout) {
-            var subscriptions = [],
-                generating = false;
-
-            //
-            function matchesSource(request) {
-                return request.source === "generator";
-            }
-
-            // Used internally; this will be repacked by doPackage
-            function generateData(request) {
-                return {
-                    key: request.key,
-                    telemetry: new SinewaveTelemetrySeries(request)
-                };
-            }
-
-            //
-            function doPackage(results) {
-                var packaged = {};
-                results.forEach(function (result) {
-                    packaged[result.key] = result.telemetry;
-                });
-                // Format as expected (sources -> keys -> telemetry)
-                return { generator: packaged };
-            }
-
-            function requestTelemetry(requests) {
-                return $timeout(function () {
-                    return doPackage(requests.filter(matchesSource).map(generateData));
-                }, 0);
-            }
-
-            function handleSubscriptions() {
-                subscriptions.forEach(function (subscription) {
-                    var requests = subscription.requests;
-                    subscription.callback(doPackage(
-                        requests.filter(matchesSource).map(generateData)
-                    ));
-                });
-            }
-
-            function startGenerating() {
-                generating = true;
-                $timeout(function () {
-                    handleSubscriptions();
-                    if (generating && subscriptions.length > 0) {
-                        startGenerating();
-                    } else {
-                        generating = false;
-                    }
-                }, 1000);
-            }
+    function SinewaveTelemetryProvider() {
```

Comment:
Two providers are being registered though - [old style](https://github.com/nasa/openmct/blob/311e221af55c0d50aa5f9f8a1464e467b23337ff/example/generator/bundle.js#L43), [new style](https://github.com/nasa/openmct/blob/311e221af55c0d50aa5f9f8a1464e467b23337ff/example/generator/src/plugin.js#L31)

I confirmed this with some testing, it's resulting in a doubling of telemetry when combined with #1348.